src/inmanta/app.py

```diff
@@ -473,10 +473,8 @@ def export(options: argparse.Namespace) -> None:
             raise CLIException(
                 "The --delete-resource-set option should always be used together with the --partial option", exitcode=1
             )
-        if not options.delete_resource_set and options.soft_delete:
-            raise CLIException(
-                "The --soft-delete option should always be used together with the --delete-resource-set option", exitcode=1
-            )
+        if not options.partial_compile and options.soft_delete:
+            raise CLIException("The --soft-delete option should always be used together with the --partial option", exitcode=1)
```

Comment:
This will not be enough to fix the issue. The lsm extension always sets `soft_delete=True`. Also when no partial compile is done.

Reply:
Hmm right, do we just remove this check ?

Reply:
Fine by me.